File: examples/python/ws_client.py
"""Send incremental text over the ElevenLabs-style WebSocket, collect audio."""

import asyncio
import base64
import json

import websockets

URL = "ws://localhost:8000/v1/text-to-speech/default/stream-input"


async def main() -> None:
    audio = b""
    msg: dict = {}
    async with websockets.connect(URL) as ws:
        for part in ["Hello, ", "welcome to our realtime voice demo.", ""]:
            await ws.send(json.dumps({"text": part}))
        while True:
            msg = json.loads(await ws.recv())
            audio += base64.b64decode(msg["audio"])
            if msg["isFinal"]:
                break
    print(f"received {len(audio)} bytes of pcm from {msg['backend']}")


if __name__ == "__main__":
    asyncio.run(main())

File: examples/javascript/ws_client.mjs
// node >= 21 (built-in WebSocket). Streams text in, collects base64 audio out.
const url = "ws://localhost:8000/v1/text-to-speech/default/stream-input";
const ws = new WebSocket(url);
let bytes = 0;

ws.onopen = () => {
  for (const text of ["Hello, ", "welcome to our realtime voice demo.", ""]) {
    ws.send(JSON.stringify({ text }));
  }
};

ws.onmessage = (event) => {
  const msg = JSON.parse(event.data);
  bytes += Buffer.from(msg.audio, "base64").length;
  if (msg.isFinal) {
    console.log(`received ${bytes} bytes of pcm from ${msg.backend}`);
    ws.close();
  }
};
